crates/bevy_window/src/windows.rs

```diff
@@ -1,36 +1,69 @@
 use super::{Window, WindowId};
 use bevy_utils::HashMap;
 
+/// A collection of [`Window`]s with unique [`WindowId`]s.
 #[derive(Debug, Default)]
 pub struct Windows {
     windows: HashMap<WindowId, Window>,
 }
 
 impl Windows {
+    /// Add the provided [`Window`] to the [`Windows`] resource.
     pub fn add(&mut self, window: Window) {
         self.windows.insert(window.id(), window);
     }
 
+    /// Get a reference to the provided [`WindowId`].
     pub fn get(&self, id: WindowId) -> Option<&Window> {
         self.windows.get(&id)
     }
 
+    /// Get a mutable reference to the provided [`WindowId`].
     pub fn get_mut(&mut self, id: WindowId) -> Option<&mut Window> {
         self.windows.get_mut(&id)
     }
 
+    /// Get a reference to the primary [`Window`].
     pub fn get_primary(&self) -> Option<&Window> {
         self.get(WindowId::primary())
     }
 
+    /// Get a reference to the primary [`Window`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if the primary window does not exist in [`Windows`]
+    pub fn primary(&self) -> &Window {
+        self.get_primary().expect("Primary window does not exist")
+    }
+
+    /// Get a mutable reference to the primary [`Window`].
     pub fn get_primary_mut(&mut self) -> Option<&mut Window> {
         self.get_mut(WindowId::primary())
     }
 
+    /// Get a mutable reference to the primary [`Window`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if the primary window does not exist in [`Windows`]
+    pub fn primary_mut(&mut self) -> &mut Window {
+        self.get_primary_mut()
+            .expect("Primary window does not exist")
+    }
+
+    /// Returns the scale factor of the primary window, or `1.0` if the window does not exist.
+    pub fn primary_scale_factor(&self) -> f64 {
```

Comment:
So, I get that this can be factored out

However, ideally nothing internally should be special for the primary window

That is to say, this method existing suggests 'relying only on the primary window's scale factor is fine'

Reply:
My justification for this is basically: 99.9% of users will only have one window. If someone is managing multiple windows, they're already in the minority and can use the existing functions bevy provides.

Reply:
Perhaps a good compromise is:
```rs
pub fn scale_factor(&self, id: &WindowId) -> f64 {
    if let Some(window) = self.get(id) {
        window.scale_factor()
    } else {
        1.0
    }
}
```

which can be used like

```rs
let scale_factor = windows.scale_factor(WindowId::primary());
```